assets/js/custom.js

```diff
@@ -1,15 +1,100 @@
-var controllerHome = (function(jQuery) {
-    var adjustHomeLink = function() {
-        if (document.location.hostname === 'localhost') {
-            jQuery('.navbar-brand').attr('href', '/');
-            jQuery('#nav-0').attr('href', '/');
-        }
-    };
-
-    return {
-        init: adjustHomeLink
-    };
-
-}(jQuery));
+var controllerHome = (function (jQuery) {
+  const CONSENT = 'consentGiven';
+  const consentBanner = jQuery("#consent-banner");
+  const acceptCookiesButton = jQuery("#accept-cookies");
+  const denyCookiesButton = jQuery("#deny-cookies");
+  const gtagId = "{{ site.gtagId }}";
+
+  var adjustHomeLink = function () {
+    if (document.location.hostname === "localhost") {
+      jQuery(".navbar-brand").attr("href", "/");
+      jQuery("#nav-0").attr("href", "/");
+    }
+  };
+
+  let initializeConsentMode = function () {
+    window.dataLayer = window.dataLayer || [];
+    function gtag() {
+      window.dataLayer.push(arguments);
+    }
+    gtag("consent", "default", {
+      ad_storage: "denied",
+      analytics_storage: "denied",
+    });
+  };
+
+  // Function to enable Google Analytics
+  let enableAnalytics = function () {
+    if (localStorage.getItem(CONSENT) === "true") {
+      window.dataLayer = window.dataLayer || [];
+      function gtag() {
+        window.dataLayer.push(arguments);
+      }
+      gtag("js", new Date());
```

Comment:
Could you also adjust the file to use single quotes whenever possible? 
```suggestion
      gtag('js', new Date());
```